fuel-block-producer/src/block_producer.rs

```diff
@@ -0,0 +1,194 @@
+use crate::{
+    block_producer::transaction_selector::select_transactions,
+    db::BlockProducerDatabase,
+    ports::{
+        Relayer,
+        TxPool,
+    },
+    Config,
+};
+use anyhow::{
+    Context,
+    Result,
+};
+use chrono::Utc;
+use fuel_core_interfaces::{
+    block_producer::{
+        BlockProducer as Trait,
+        Error::{
+            GenesisBlock,
+            InvalidDaFinalizationState,
+            MissingBlock,
+        },
+    },
+    common::{
+        crypto::ephemeral_merkle_root,
+        fuel_tx::CheckedTransaction,
+        fuel_types::Bytes32,
+    },
+    executor::{
+        Error,
+        ExecutionMode,
+        Executor,
+    },
+    model::{
+        BlockHeight,
+        DaBlockHeight,
+        FuelBlock,
+        FuelBlockHeader,
+    },
+};
+use tracing::{
+    debug,
+    error,
+};
+
+#[cfg(test)]
+mod tests;
+mod transaction_selector;
+
+pub struct Producer<'a> {
+    pub config: Config,
+    pub db: &'a dyn BlockProducerDatabase,
+    pub txpool: &'a dyn TxPool,
+    pub executor: &'a dyn Executor,
+    pub relayer: &'a dyn Relayer,
```

Comment:
Do we need to put it under `feature = "relayer"`?

Reply:
I think it would be simpler for this module if fuel-core just passed in a dummy relayer implementation when the feature is disabled. Since it's behind a trait, the behavior is already configurable.